sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/QueryResult.cs

```diff
@@ -54,6 +52,21 @@ internal static QueryResult<T> DeserializeQueryResult(JsonElement element, Objec
         {
             IReadOnlyList<T> items = default;
             string continuationToken = default;
+
+            // If the provided objectSerializer is of type JsonObjectSerializer, we will re-use the same object and set it as the defaultObjectSerializer.
+            if (objectSerializer is JsonObjectSerializer)
```

Comment:
This is not thread safe. Perhaps do this during client initialization instead.

Reply:
you are right, this is not threadsafe. I will move the instantiation to the client.